common/src/main/scala/org/neo4j/spark/writer/BaseDataWriter.scala

```diff
@@ -91,15 +94,27 @@ abstract class BaseDataWriter(jobId: String,
     Unit
   }
 
+  /**
+   * df: we check if the thrown exception is STOPPED_THREAD_EXCEPTION. This is the
+   * exception that is thrown when the streaming query is interrupted, we don't want to cause
+   * any error in this case. The transaction are rolled back automatically.
+   */
   private def logAndThrowException(e: Exception): Unit = {
-    if (e.isInstanceOf[ClientException]) {
-      log.error(s"Cannot commit the transaction because: ${e.getMessage}")
+    if(e.isInstanceOf[ServiceUnavailableException] && e.getMessage == STOPPED_THREAD_EXCEPTION_MESSAGE) {
+      // df: leaving it here so we make sure if the cause is an instance of TaskKilledException
+      logInfo(s"+++ Cause: ${e.getCause.toString}")
+      logInfo(e.getMessage)
     }
     else {
-      log.error("Cannot commit the transaction because the following exception", e)
-    }
+      if ((e.isInstanceOf[ServiceUnavailableException] && e.getMessage == STOPPED_THREAD_EXCEPTION_MESSAGE) || e.isInstanceOf[ClientException]) {
```

Comment:
(e.isInstanceOf[ServiceUnavailableException] && e.getMessage == STOPPED_THREAD_EXCEPTION_MESSAGE)
this is always false in this section right?

Reply:
yep